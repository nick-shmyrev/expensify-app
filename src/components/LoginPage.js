import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth-actions';

export const LoginPage = props => (
  <div>
    <button name="btn-login" onClick={props.startLogin}>Login</button>
  </div>
);

const mapDispatchToProps = dispatch => ({
  startLogin: () => dispatch(startLogin()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);

