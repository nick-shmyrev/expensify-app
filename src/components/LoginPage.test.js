import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from './LoginPage';

describe('LoginPage component', () => {
  const startLogin = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={startLogin}/>);
  
  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  test('should call startLogin on button click', () => {
    wrapper.find('button[name="btn-login"]').simulate('click');
    
    expect(startLogin).toHaveBeenCalled();
  });
});
