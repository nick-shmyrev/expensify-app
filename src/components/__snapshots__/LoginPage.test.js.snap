// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component should render correctly 1`] = `
<div>
  <button
    name="btn-login"
    onClick={[MockFunction]}
  >
    Login
  </button>
</div>
`;
